hudi-common/src/main/java/org/apache/hudi/metadata/HoodieBackedTableMetadata.java

```diff
@@ -824,7 +825,8 @@ public Map<String, Set<String>> getSecondaryIndexRecords(List<String> keys, Stri
       return Collections.emptyMap();
     }
 
-    return getRecordsByKeyPrefixes(keys, partitionName, false).map(
+    return getAllRecordsByKeys(keys, partitionName).values().stream().flatMap(Collection::stream)
+        .map(
```

Comment:
Since the file group mapping changes from table version 8 to 9, we should differentiate the reading and look up logic based on the table version, to avoid wrong results (note that the secondary index layout changes on storage).